src/instructlab/config.py

```diff
@@ -216,3 +223,50 @@ def get_model_family(forced, model_path):
     guess = MODEL_FAMILY_MAPPINGS.get(guess, guess)
 
     return guess if guess in MODEL_FAMILIES else DEFAULT_MODEL_FAMILY
+
+class Lab:
+    """Lab object holds high-level information about ilab CLI"""
+
+    def __init__(self, config_obj: Config):
+        self.config = config_obj
+
+        # Set up logging for the Lab class
+        self.logger = logging.getLogger(__name__)
+
+        # Create a formatter
+        formatter = log.CustomFormatter(log.FORMAT)
+
+        # Create a handler and set the formatter
+        handler = logging.StreamHandler()
+        handler.setFormatter(formatter)
+
+        logging.basicConfig(
+            format=log.FORMAT,
+            handlers=[handler],
+        )
+
+        self.logger.setLevel(self.config.general.log_level.upper())
+
+    
+
+def init_config(ctx, config_file):
```

Comment:
is this a blocker to merge this PR? I think init_config is fine and more descriptive IMO and this would create an unnecessary rebase on all the commits.

Reply:
i don't think this is a blocker but it would be good to create an issue for this.
having `config.init_config()` does indicate something off with the abstraction.

Reply:
https://github.com/instructlab/instructlab/issues/1323